### ISSUE TITLE
Automatically detect the java version each Minecraft version needs upon instance creation.

### ISSUE THREAD

**Issue by Ekain06** [MultiMC/MultiMC5#4013]
### Role

I want to create 1.17+ instances as well as playing modpacks and such.

### Suggestion

I want MultiMC to automatically assign the correct java version for each instance at creation.

### Benefit

So that I don't need to fiddle with settings when creating a new instance, and so that new users don't get confused when Minecraft doesn't launch.

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

MultiMC can already detect the different JREs installed including each's version. Using this and the newly added "javaVersion" key to the vanilla version JSONs, I suppose it wouldn't be *too* difficult to implement. (I don't know MultiMCs codebase so I could be wrong here)

**Comment by MythicBlade** [comment]
I think this is an amazing Idea this is something that has messed me up before

**Comment by bryiewes** [comment]
i like this idea, and i feel like it could in theory be easy to implement, but its minor imo

**Comment by Zetabite** [comment]
Its not minor if it is one of the main cases of people having issues with MMC

**Comment by jredfox** [comment]
yes but edit it to be optional. like auto select the recommended jvm, then have it automatically download to a path you give it. this is to prevent it from being actually installed interfering with the main jvm of the end user's main java installation.

**Comment by cootshk** [comment]
do it sort of like the modrinth launcnher does

https://github.com/PrismLauncher/PrismLauncher/issues/1720 (see the screenshot)

**Comment by ascpial** [comment]
This is even more needed as mojang added [yet another version of java to support](https://www.minecraft.net/en-us/article/minecraft-snapshot-24w14a).
